samza-test/src/test/java/org/apache/samza/storage/kv/TransactionalStateIntegrationTest.java

```diff
@@ -116,35 +116,53 @@ public void setUp() {
   @Test
   public void testStopAndRestart() {
     List<String> inputMessagesOnInitialRun = Arrays.asList("1", "2", "3", "2", "97", "-97", ":98", ":99", ":crash_once");
+    // double check collectors.flush
     List<String> expectedChangelogMessagesOnInitialRun = Arrays.asList("1", "2", "3", "2", "97", null, "98", "99");
     initialRun(inputMessagesOnInitialRun, expectedChangelogMessagesOnInitialRun);
 
-    // first two are reverts for uncommitted messages from last run
-    List<String> expectedChangelogMessagesOnSecondRun =
+    // first two are reverts for uncommitted messages from last run for keys 98 and 99
+    List<String> expectedChangelogMessagesAfterSecondRun =
         Arrays.asList(null, null, "98", "99", "4", "5", "5");
     List<String> expectedInitialStoreContentsOnSecondRun = Arrays.asList("1", "2", "3");
     secondRun(CHANGELOG_TOPIC,
-        expectedChangelogMessagesOnSecondRun, expectedInitialStoreContentsOnSecondRun);
+        expectedChangelogMessagesAfterSecondRun, expectedInitialStoreContentsOnSecondRun, CONFIGS);
+  }
+
+  @Test
+  public void testStopAndRestartCheckpointV2() {
```

Comment:
Since this is an unrelated scenario, can you extract this to a new class (CheckpointVersionIntegrationTest?).